### ISSUE TITLE
object-curly-spacing possible bug with destructuring and trailing commas

### ISSUE THREAD

**Issue by lencioni** [eslint/eslint#2647]
I have the following code:

```js
const testing = { test: 1 };
let {
  test,
} = testing;
                               
test++;
```

With the new object-curly-spacing rule set to `always` in ESLint 0.22.1, I see the following output:

> 3:6  error  A space is required before ','  object-curly-spacing

I wasn't able to find any mention of this in the documentation for the rule, so I'm not sure if it was intentional or not, but it seems a bit strange to me so I thought it might be a bug.

Thanks for all of your great work on this tool!

**Comment by ilyavolodin** [comment]
I just verified it, and it does error out. However, I think the error is correct, the message is wrong though. I think it's erroring on a let statement, because with `always` keyword, you have to have a space after a comma. Now the message is saying that space is missing before the comma, so the message is clearly wrong.

**Comment by nzakas** [comment]
Isn't this rule specifically about the curlies? Why would it care about commas at all?

**Comment by xjamundx** [comment]
I assume that's a typo, but will double check.

**Comment by xjamundx** [comment]
Here's what I get with `0.22.1`:

```
  1:16  error  There should be no space after '{'   object-curly-spacing
  1:26  error  There should be no space before '}'  object-curly-spacing
```

**Comment by xjamundx** [comment]
We use `last = context.getTokenAfter(lastSpecifier);` because we don't handle the trailing comma. When we pass this to the reporter, it just reads the value for that token, which in our minds will always be `}` or ` `, but in this case it's `,`

**Comment by ilyavolodin** [comment]
Hmmm.. I don't think adding explicit support for trailing commas would work. What about something like this:
```js
{ name: something ? something : other }
```
or
```js
{ name: name || "empty" }
```
It feels like we should find closing brace and check if the previous character is space.

**Comment by xjamundx** [comment]
I put together a quick PR which does it the way I described, feel free to comment in there with any feedback which I'll gladly apply

**Comment by ilyavolodin** [comment]
Nevermind, I just checked the examples I wrote, and they all work fine.
